mysql_mimic/intercept.py

```diff
@@ -0,0 +1,49 @@
+from __future__ import annotations
+
+from typing import Any
+
+from sqlglot import expressions as exp
+from mysql_mimic.variables import DEFAULT
+
+
+def setitem_kind(setitem: exp.SetItem) -> str:
+    kind = setitem.text("kind")
+    if not kind:
+        return "VARIABLE"
+
+    if kind in {"GLOBAL", "PERSIST", "PERSIST_ONLY", "SESSION", "LOCAL"}:
+        return "VARIABLE"
+
+    return kind
+
+
+def value_to_expression(value: Any) -> exp.Expression:
+    if value is True:
+        return exp.TRUE.copy()
+    if value is False:
+        return exp.FALSE.copy()
+    if value is None:
+        return exp.NULL.copy()
+    if isinstance(value, (int, float)):
+        return exp.Literal.number(value)
+    return exp.Literal.string(str(value))
+
+
+def expression_to_value(expression: exp.Expression) -> Any:
+    if expression == exp.TRUE:
```

Comment:
i guess this is ok to use the singletons for just checks?

Reply:
Ah, nice catch. Been working on this one for a while, since before we added those new functions.

Reply:
(I think it's better to use the functions, even if its slightly less efficient?)

Reply:
yea